### PR TITLE
Briankang pytorch torchrun

### PR DESCRIPTION
**Pull request by brianchunkang**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
Adding notebook tutorial for running training with PyTorch torchrun with custom service accounts
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [x] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb [hunk 1]
```diff
@@ -0,0 +1,2732 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Title: sentence case and branding:</p><h1>PyTorch multi-node distributed training with Torchrun on Vertex AI Training using a custom container </h1><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Split cell into 4: Overview, Objective, Dataset, Costs. Note that Dataset is after Objective.</p><p> </p><p>Google Cloud Storagte -&gt; Cloud Storage</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Make installs a single pip3</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Google Artifact Registry -&gt;  Artifact Registry.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Vertex Training -&gt; Vertex AI Training</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>combine code cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (andrewferlitsch) [inline notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>remove TODO</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1323/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='77'/>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>There are two docker containers, one for the host and one for the workers</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>There are two docker containers, one for the host and one for the workers. At one point the main.py was slightly different, and the main.sh is definitely different</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>updated</p>

[the remark this answers is not in the document]
Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>done</p>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Updated</p>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Updated</p>

Comment (brianchunkang) [reply notebooks/community/sdk/sdk_pytorch_torchrun_custom_container_training_imagenet.ipynb:1]: <p>Moved section and updated Google Cloud Storage to Cloud Storage</p>

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**